examples/python/nuscenes/main.py

```diff
@@ -0,0 +1,218 @@
+            points = pointcloud.points[:3].T  # shape after transposing: (num_points, 3)
+            point_distances = np.linalg.norm(points, axis=1)
+            point_colors = cmap(norm(point_distances))
+            rr.log(f"world/ego_vehicle/{sensor_name}", rr.Points3D(points, colors=point_colors))
+            current_camera_token = sample_data["next"]
+
+
+def log_annotations(first_sample_token: str, start_timestamp: numbers.Number, nusc: nuscenes.NuScenes) -> None:
+    """Log 3D bounding boxes."""
+    label2id: dict[str, int] = {}
+    current_sample_token = first_sample_token
+    while current_sample_token != "":
+        sample = nusc.get("sample", current_sample_token)
+        rr.set_time_seconds("timestamp", (sample["timestamp"] - start_timestamp) * 1e-6)
+        ann_tokens = sample["anns"]
+        sizes = []
+        centers = []
+        rotations = []
+        class_ids = []
+        for ann_token in ann_tokens:
+            ann = nusc.get("sample_annotation", ann_token)
+
+            rotation_xyzw = np.roll(ann["rotation"], shift=-1)
+            width, length, height = ann["size"]
+            sizes.append((length, width, height))  # x, y, z sizes
+            centers.append(ann["translation"])
+            rotations.append(rr.Quaternion(xyzw=rotation_xyzw))
+            if ann["category_name"] not in label2id:
+                label2id[ann["category_name"]] = len(label2id)
+            class_ids.append(label2id[ann["category_name"]])
+
+        rr.log("world/anns", rr.Boxes3D(sizes=sizes, centers=centers, rotations=rotations, class_ids=class_ids))
+        current_sample_token = sample["next"]
+
+
+def log_sensor_calibration(sample_data: dict[str, Any], nusc: nuscenes.NuScenes) -> None:
+    """Log sensor calibration (pinhole camera, sensor poses, etc.)."""
+    sensor_name = sample_data["channel"]
+    calibrated_sensor_token = sample_data["calibrated_sensor_token"]
+    calibrated_sensor = nusc.get("calibrated_sensor", calibrated_sensor_token)
+    rotation_xyzw = np.roll(calibrated_sensor["rotation"], shift=-1)
+    rr.log(
+        f"world/ego_vehicle/{sensor_name}",
+        rr.Transform3D(
+            translation=calibrated_sensor["translation"],
+            rotation=rr.Quaternion(xyzw=rotation_xyzw),
+            from_parent=False,
+        ),
+        timeless=True,
+    )
+    if len(calibrated_sensor["camera_intrinsic"]) != 0:
+        rr.log(
+            f"world/ego_vehicle/{sensor_name}",
+            rr.Pinhole(
+                image_from_camera=calibrated_sensor["camera_intrinsic"],
+                width=sample_data["width"],
+                height=sample_data["height"],
+            ),
+            timeless=True,
+        )
+
+
+def main() -> None:
+    parser = argparse.ArgumentParser(description="Visualizes the nuScenes dataset using the Rerun SDK.")
+    parser.add_argument(
+        "--root_dir",
+        type=pathlib.Path,
+        default="dataset",
```

Comment:
Would it make sense for this dataset dir to default to <script dir>/dataset ?

Reply:
The examples with extra modules only run when in the script dir any way (same for arkit_scenes and raw_mesh) due to the import of the other module.

Reply:
Fixed it.